src/common/controllers/ads-controller.js

```diff
@@ -280,11 +291,12 @@ class AdsController extends FakeEventTarget implements IAdsController {
 
   _onAdLoaded(event: FakeEvent): void {
     this._adIsLoading = false;
-    this._ad = event.payload.ad;
+    this._pendingAd = event.payload.ad;
```

Comment:
@dan-ziv same issue as I mentioned in playkit, postroll IMA+BUMPER postroll.
you'll get the incorrect pending ad since IMA post-roll loads before bumper, on bumper ad start playing `getPendingAd` will return null and needs to return IMA postroll.

Reply:
Just because I added `pendingAd` this issue will reflect in an API call. But it's a separate issue that has been around all along: you'll trigger AD_LOADED event twice with a different ad payload.
So, anyway, I think we need to handle it in a different ticket and not overload it here. We can remove in the meanwhile the `pendingAd` API to avoid this issue. @OrenMe @yairans what do u think?

Reply:
Agree, if you keep the `pendingAd` API it should be fixed since it won't return the correct value.

Reply:
I would remove it as no one is using it

Reply:
So agreed - let's remove from here and open ticket to handle the next ad issue.
I think it is ok to have a nextAd object with this exact same logic as you added here.
This way we have a flow where when ad is loaded you can get it in getnextAd and when it starts we clear nextAd and update the _ad object that can be retrieved in getAd.
And when another ad get's loaded it will be populated in the _nextAd - this can happen in several ways, if this is an ad pod then while current ad is playing the next will be loaded or for IMA specifically when last midroll finish and there is a postroll the postroll is already loaded.